### PR TITLE
[Alter]Clean SchemaChangeJobV2 when schema change CANCELLED or FINISHED

### PR DESCRIPTION
**Pull request by WingsGo**
SchemaChangeJobV2 will use too much memory in FE, which may casuse FullGC. But these data is useless after job is done, so we need to clean it up.

### CODE REVIEW

**Comment by WingsGo** [comment]
For #3132

File: fe/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java [hunk 1]
```diff
@@ -953,4 +969,131 @@ public void readFields(DataInput in) throws IOException {
             }
         }
     }
+
+    /**
+     * write data need to persist when job finished
+     */
+    private void writeJobFinishedData(DataOutput out) throws IOException {
+        // only persist data will be used in getInfo
+        out.writeInt(indexIdMap.size());
```
Comment (kangkaisen) [inline fe/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java]: Write double?

File: fe/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java [hunk 2]
```diff
@@ -953,4 +969,131 @@ public void readFields(DataInput in) throws IOException {
...
+            for (int i = 0; i < bfNum; i++) {
+                bfColumns.add(Text.readString(in));
+            }
+            bfFpp = in.readDouble();
+        }
+
+        watershedTxnId = in.readLong();
+
+        // index
+        if (Catalog.getCurrentCatalogJournalVersion() >= FeMetaVersion.VERSION_70) {
```
Comment (kangkaisen) [inline fe/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java]: Don't need this check.

File: fe/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java [hunk 3]
```diff
@@ -95,6 +95,9 @@
     // shadow index id -> shadow index short key count
     private Map<Long, Short> indexShortKeyMap = Maps.newHashMap();
 
+    // identify whether the job is finished and no need to persist some data
+    private boolean cleared = false;
```
Comment (kangkaisen) [inline fe/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java]: Please rename. Such as `isMetaPruned `

File: fe/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java [hunk 4]
```diff
@@ -159,6 +161,18 @@ public void setStorageFormat(TStorageFormat storageFormat) {
         this.storageFormat = storageFormat;
     }
 
+    /**
+     * clear some date structure in this job to save memory
+     * these data structures must not used in getInfo method
+     */
+    public void clear() {
```
Comment (kangkaisen) [inline fe/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java]: Please rename. Such as `pruneMeta`

File: fe/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java (hunk 1 above)
Comment (WingsGo) [reply fe/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java]: why use writeDouble? indexIdMap.size() is a integer type.

File: fe/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java (hunk 4 above)
Comment (WingsGo) [reply fe/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java]: done

File: fe/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java (hunk 2 above)
Comment (WingsGo) [reply fe/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java]: I just use the origin code, why we don't need this check?

File: fe/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java (hunk 1 above)
Comment (kangkaisen) [reply fe/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java:978]: Ignore me. I yesterday looked the code write `out.writeInt(indexIdMap.size());` twice. So I said write double.

File: fe/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java (hunk 2 above)
Comment (kangkaisen) [reply fe/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java:1052]: We only call `readJobFinishedData` If `Catalog.getCurrentCatalogJournalVersion() >= FeMetaVersion.VERSION_80` is true.

Comment (WingsGo) [reply fe/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java]: done

**Review by kangkaisen — APPROVED** [review]
+1， LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**